components/dashboard/src/workspaces/PersonalizedContent.tsx

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import React, { useEffect, useState } from "react";
+import { User } from "@gitpod/public-api/lib/gitpod/v1/user_pb";
+import { useCurrentUser } from "../user-context";
+
+type ContentItem = {
+    url: string;
+    title: string;
+    label: string;
+};
+
+const contentList: ContentItem[] = [
+    {
+        url: "https://www.gitpod.io/blog/writing-software-with-chopsticks-an-intro-to-vdi",
+        title: "Why replace your VDI with CDE",
+        label: "vdi-replacement",
+    },
+    {
+        url: "https://www.gitpod.io/customers/luminus",
+        title: "Solve python dependency issues with Gitpod",
+        label: "luminus-case-study",
+    },
+    {
+        url: "https://www.gitpod.io/blog/how-to-use-vdis-and-cdes-together",
+        title: "Using VDIs and Gitpod together",
+        label: "vdi-and-cde",
+    },
+    {
+        url: "https://www.gitpod.io/blog/onboard-contractors-securely-and-quickly-using-gitpod",
+        title: "Onboard contractors with Gitpod",
+        label: "onboard-contractors",
+    },
+    {
+        url: "https://www.gitpod.io/solutions/onboarding",
+        title: "Onboarding developers in one click",
+        label: "onboarding-solutions",
+    },
+    {
+        url: "https://www.gitpod.io/customers/kingland",
+        title: "How Gitpod impacts supply chain security",
+        label: "kingland-case-study",
+    },
+    {
+        url: "https://www.gitpod.io/blog/improve-security-using-ephemeral-development-environments",
+        title: "Improve security with ephemeral environments",
+        label: "ephemeral-security",
+    },
+    {
+        url: "https://www.gitpod.io/blog/using-a-cde-roi-calculator",
+        title: "Building a business case for Gitpod",
+        label: "cde-roi-calculator",
+    },
+    {
+        url: "https://www.gitpod.io/blog/whats-a-cloud-development-environment",
+        title: "What's a CDE",
+        label: "what-is-cde",
+    },
+];
+
+const defaultContent: ContentItem[] = [
+    {
+        url: "https://www.gitpod.io/blog/whats-a-cloud-development-environment",
+        title: "What's a CDE",
+        label: "what-is-cde",
+    },
+    {
+        url: "https://www.gitpod.io/solutions/onboarding",
+        title: "Onboarding developers in one click",
+        label: "onboarding-solutions",
+    },
+    {
+        url: "https://www.gitpod.io/blog/using-a-cde-roi-calculator",
+        title: "Building a business case for Gitpod",
+        label: "cde-roi-calculator",
+    },
+];
+
+const PersonalizedContent: React.FC = () => {
+    const user: User | undefined = useCurrentUser();
+    const [selectedContent, setSelectedContent] = useState<ContentItem[]>([]);
+
+    useEffect(() => {
+        const storedContentData = localStorage.getItem("personalized-content-data");
```

Comment:
`localStorage` can throw. We have a method to check for possible storage issues:

https://github.com/gitpod-io/gitpod/blob/2a8ee81ca3ffbcd3809e7ab967ef3b90a40b8f3d/components/dashboard/src/utils.ts#L123-L153

Reply:
Resolved in [ed6ac63](https://github.com/gitpod-io/gitpod/pull/20034/commits/ed6ac63638937746281b17ab1d525ee39e13e5f8)